CompanyPaymentsProject/Company.cs

```diff
@@ -6,41 +6,30 @@
 
 namespace CompanyPaymentsProject
 {
-    class Company : Person
+    class Company
     {
-        
-        //lista osób i możliwość ich dodawania
-        List<Person> people = new List<Person>()
+        private List<Person> _persons = new List<Person>();
+       
+        public List<Person> GetPersonList()
         {
-            new Person { PersonID = 1, Name = "Jan", Surname = "Kowalski", Post.Assistant},
-            new Person { PersonID = 2, Name = "Anna", Surname = "Nowak", Post.Junior},
-            new Person { PersonID = 3, Name = "John", Surname = "Doe", Post.Manager},
-        };
-        public void AddNewWorker()
+            return _persons;
+        }
+
+        public void AddPayCheck(int personId)
         {
-            for (int i = people.Count; i < people.Count+1; i++)
-            {
-                people[i] = new Person();
-                Console.WriteLine("Enter worker #{0} name", i + 1 + people.Count);
-                people[i].Name = Console.ReadLine();
-                Console.WriteLine("Enter worker #{0} Surname", i + 1 + people.Count);
-                people[i].Surname =Console.ReadLine();
-                Console.WriteLine("Set the worker post in the company: 1. Manager, 2. Junior, 3. Assistant");
-                int workerPost = int.Parse(Console.ReadLine());
+            var person = _persons.Where(x => x.PersonID == personId);            
```

Comment:
To jak się głębiej zastanowię to chyba się niezbyt kompiluje ;). Skoro używasz id to musisz uzyskać unikalną osobę, więc użyj tutaj SingleOrDefault(x => x.PersonId == personId);
Zadziała tak jak Ci odpisywałem, jeśli znajdzie to osobę zwróci, jeśli nie to zwróci null, a jeśli znajdzie więcej niż jedną osobę to coś okropnego dzieje się z danymi i dostaniesz exception